### PR TITLE
Add some good stuff

### LINKED ISSUE #81: Unique serial number
Generate the serial number based on STM UID

### PR DESCRIPTION
**Pull request by kbeckmann**
Closes #81

### CODE REVIEW

**Comment by kbeckmann** [comment]
Is the stuff good enough?

File: app/drivers/usb.c [hunk 1]
```diff
@@ -251,6 +250,8 @@ static uint8_t desc_device_qualifier[] =
 	0x00,
 };
 
+static char usb_serialnumber_string[13];
```
Comment (arturo182) [inline app/drivers/usb.c]: Why not in `self`?

File: app/errorhandler.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+{
+	HardFault_Handler();
+}
+
+void BusFault_Handler(void)
+{
+	HardFault_Handler();
+}
+
+void assert_failed(uint8_t* file, uint32_t line)
```
Comment (arturo182) [inline app/errorhandler.c]: Move * to right side of the variable

Comment (arturo182) [reply app/errorhandler.c]: Also maybe p_

File: app/drivers/usb.c (hunk 1 above)
Comment (kbeckmann) [reply app/drivers/usb.c]: Lol, there were so many separate static variables that I missed that we had already created a `me`.

**Comment by kbeckmann** [comment]
Added platform.c/h in order to have a portable platform_delay_us() that we can implement on any platform.

File: app/drivers/usb.c (hunk 1 above)
Comment (arturo182) [reply app/drivers/usb.c]: You mean a `self`, there's no `me` here, not sure what you're talking about :P

Comment (kbeckmann) [reply app/drivers/usb.c]: lol. I am not sure where this concept of `me` comes from either, never heard about it. Anyway, I moved it to `self` and seems to work.

**Comment by kbeckmann** [comment]
Umm.. My device's serial is 001b005c5246500c. That's a lot of 0s, what could be wrong?

**Comment by arturo182** [comment]
Only parts of the UID are actually unique per device, many others are family and manuf. time based. The whole UID is unique but parts might be not. So yeah, we could xor the words together or something.

**Comment by kbeckmann** [comment]
Updated the patch, ready to merge (using cherrypick this time please)

**Pull request merged**